pkg/dab/module.go

```diff
@@ -15,6 +19,164 @@ const (
 	MagicFilename_Plot   = "plot.wf"
 )
 
+// See validateDNS1123Subdomain and ValidateModuleName
+const (
+	// similar to dns1123 label hunks, but allows mid-string dots also.
+	validation_moduleNamePathHunk_regexpStr string = "[a-z0-9]([-a-z0-9\\.]*[a-z0-9])?"
+	validation_moduleNamePathHunk_msg       string = "must consist of lower case alphanumeric characters or '-' or '.', and must start and end with an alphanumeric character"
+	validation_moduleNamePathHunk_maxlen    int    = 63
+	validation_dns1123Label_regexpStr       string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"
+	validation_dns1123Label_msg             string = "must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character"
+	validation_dns1123Label_maxlen          int    = 63
+	validation_dns1123Subdomain_regexpStr   string = validation_dns1123Label_regexpStr + "(\\." + validation_dns1123Label_regexpStr + ")*"
+	validation_dns1123Subdomain_msg         string = "must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character"
+	validation_dns1123Subdomain_maxlen      int    = 253
+)
+
+var (
+	validation_moduleNamePathHunk_regexp = regexp.MustCompile("^" + validation_moduleNamePathHunk_regexpStr + "$")
+	validation_dns1123Subdomain_regexp   = regexp.MustCompile("^" + validation_dns1123Subdomain_regexpStr + "$")
+)
+
+// ValidateModuleName checks the module name for invalid strings.
```

Comment:
Maybe add some examples off valid module names in this comment.

Reply:
Hmm... perhaps. I'm not convinced it will add value but it's not a lot of effort and I need to change this doc string anyway.
I definitely need to remove the line about underscores being allowed since that's no longer true!

Reply:
If there's anything I value more and more over the years, it's examples.  :)

If in the docs doesn't feel robust enough, the next two steps up are: golang's `Example_` test functions are pretty neat for getting things into godoc; or, making a quick invocation of testmark and putting it in our markdown docs directories is also almost always great.  (I'd say in this case, the latter is more likely to be useful -- we'll probably expect end users to want to see examples of strings that are recommended for use here, and they won't all be godoc consumers.)